pkg/auth/auth_flow_orchestrator.go

```diff
@@ -0,0 +1,145 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/service"
+	"github.com/flyteorg/flytestdlib/logger"
+
+	"github.com/pkg/browser"
+	"golang.org/x/oauth2"
+)
+
+const (
+	Timeout     = 15 * time.Second
+	RefreshTime = 5 * time.Minute
+)
+
+type TokenOrchestrator struct {
+}
+
+// This is a copy of oauth2.internal.tokenJSON as its not accesible outside.
+// This class is required since the json data returned by admin is in this internal format and requires to be converted to
+// the oauth2.Token format
+
+type tokenJSON struct {
+	AccessToken  string `json:"access_token"`
+	TokenType    string `json:"token_type"`
+	RefreshToken string `json:"refresh_token"`
+	ExpiresIn    int    `json:"expires_in"`
+}
+
+func (e *tokenJSON) expiry() (t time.Time) {
+	if v := e.ExpiresIn; v != 0 {
+		return time.Now().Add(time.Duration(v) * time.Second)
+	}
+	return
+}
+
+func (f TokenOrchestrator) RefreshTheToken(ctx context.Context, clientConf *oauth2.Config, token *oauth2.Token) *oauth2.Token {
+	// ClientSecret is empty here. Basic auth is only needed to refresh the token.
+	client := newBasicClient(clientConf.ClientID, clientConf.ClientSecret)
+	payload := url.Values{
+		"grant_type":    {"refresh_token"},
+		"refresh_token": {token.RefreshToken},
+		"scope":         {"all", "offline"},
+	}
+	_, body, err := client.Post(clientConf.Endpoint.TokenURL, payload)
```

Comment:
can't we use oauth2.Config to Exchange and refresh tokens?

Reply:
Removed this logic now and uses
```func (s *reuseTokenSource) Token() (*Token, error) {
```